src/libstd/ffi/c_str.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright 2012 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use fmt;
+use iter::IteratorExt;
+use libc;
+use mem;
+use ops::Deref;
+use slice::{mod, SliceExt, AsSlice};
+use string::String;
+use vec::Vec;
+
+/// A type representing a C-compatible string
+///
+/// This type serves the primary purpose of being able to generate a
+/// C-compatible string from a Rust byte slice or vector. An instance of this
+/// type is a static guarantee that the underlying bytes contain no interior 0
+/// bytes and the final byte is 0.
+///
+/// A `CString` is created from either a byte slice or a byte vector. After
+/// created, a `CString` predominately inherits all of its methods from the
+/// `Deref` implementation to `[libc::c_char]`. Note that the underlying array
+/// is represented as an array of `libc::c_char` as opposed to `u8`. A `u8`
+/// slice can be obtained with the `as_bytes` method. Note that all slices
```

Comment:
Slices produced from a `CString` do **not* contain the trailing nul terminator unless otherwise specified.